io/io/test/TFileMergerTests.cxx

```diff
@@ -101,3 +102,39 @@ TEST(TFileMerger, MergeSingleOnlyListed)
    ASSERT_TRUE(output.get() && output->GetListOfKeys());
    EXPECT_EQ(output->GetListOfKeys()->GetSize(), 2);
 }
+
+// https://github.com/root-project/root/issues/14558 aka https://its.cern.ch/jira/browse/ROOT-4716
+TEST(TFileMerger, MergeBranches)
+{
+   TTree atree("atree", "atitle");
+   int value;
+   atree.Branch("a", &value);
+  
```

Comment:
```
   TTree a2tree("abtree", "abtitle");
   a2tree.Branch("a", &value);
   value = 11;
   a2tree.Fill();
```
will show the problem with number of entry per branches.
(i.e. GetEntry(0) will erroneously return a==11 and b==42)